lib/tasks/pubmed.rake

```diff
@@ -10,4 +10,36 @@ namespace :pubmed do
     doc = client.fetch_records_for_pmid_list(pmids)
     puts doc # XML document
   end
+
+  desc 'Harvest using a text file with a list of publications by PubmedID for the supplied cap_profile_id'
+  task :pmid_profile_id_import, [:path_to_report,:cap_profile_id] => :environment do |_t, args|
+    lines = IO.readlines args[:path_to_report]
+    author = Author.find_by_cap_profile_id(args[:cap_profile_id])
+
+    puts "Cap_profile_id #{args[:cap_profile_id]} is #{author.first_name} #{author.last_name}"
+    total_pub_count=author.contributions.size
+    new_pub_count=author.contributions.where(:status=>'new').size
+    approved_pub_count=author.contributions.where(:status=>'approved').size
+    puts "total publications: #{total_pub_count}"
+    puts "total new publications: #{new_pub_count}"
+    puts "total accepted publications: #{approved_pub_count}"
+    puts "attempting to harvest #{lines.count} new publications by pubmedID"
+    puts
```

Comment:
personally i would leave the load on SQL to do the joins and counts and count on its own internal intelligence/caching instead of pulling everything into a ruby array object and iterating there...pragmatically its not likely to matter much either way since any one given author is unlikely to have a large enough number of contributions to cause a problem either way